### DISCUSSION TITLE
Problem with spatie/laravel-tags

### DISCUSSION THREAD

**Discussion by quiquegarcia** [Laravel-Backpack/community-forum#309] [Q&A (Help)]
Someone has tried to make a CRUD panel for a Tag Model extending Spatie\Tags\Tag?

To create a CRUD I needed to use CRUDTrait, so I extended from ```Spatie\Tags\Tag``` Model:

App\Models\Tag.php
```php
namespace App\Models;

use Backpack\CRUD\app\Models\Traits\CrudTrait;
use Spatie\Tags\Tag as SpatieTagModel;

class Tag extends SpatieTagModel
{
    use CrudTrait;
}
```

And set it up my CRUD Controller:

app\Http\Controllers\Admin\TagCrudController.php
```php
protected function setupCreateOperation()
{
    CRUD::setValidation(TagRequest::class);

    $this->crud->addFields([
        ['name' => 'name', 'label' => __('labels.name'), 'type' => 'text'],
        ['name' => 'slug', 'label' => __('labels.slug'), 'type' => 'text'],
        [
            'name' => 'type', 'label' => __('labels.type'), 'type' => 'select2_from_array',
            'options' => ['category' => __('labels.category'), 'topic' => __('labels.topic')],
        ],
    ]);
}

protected function setupUpdateOperation()
{
    $this->setupCreateOperation();
}
```

This resulted in a properly rendered CRUD Panel
<img width="595" alt="image" src="https://user-images.githubusercontent.com/286159/99483397-b9a31900-2923-11eb-8ee1-f877551cdbd6.png">

But when I try to create or update a Tag, translated attributes like **name** becomes null:
```sql
SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'slug' cannot be null (SQL: update `tags` set `name` = {"es":null}, `slug` = ?, `tags`.`updated_at` = 2020-11-17 22:25:27 where `id` = 13) 
```

I noticed that when I manually update the form with the input attribute ```name="name"``` to ```name="name[es]"``` update and creation works well.

This can be a bug or I'm missing something?

**Comment by pxpm** [comment]
Hello @quiquegarcia 

I think I didn't fully understand, problem is with `slug` can't be null, but you solved it changing `name` in html ?

I think the problem here relies on the slug beeing translatable. 

Are you using backpack slug traits ? 

```php
use Backpack\CRUD\app\Models\Traits\SpatieTranslatable\Sluggable;
use Backpack\CRUD\app\Models\Traits\SpatieTranslatable\SluggableScopeHelpers;
```

https://backpackforlaravel.com/docs/4.1/crud-operation-update#translatable-models-and-multi-language-cruds

Best,
Pedro

**Reply by quiquegarcia** [reply]
Hi @pxpm 

Yes, solved when I temporally change ```name``` in html.

Slug is failing because the value of name property in the model is null so Slugger is trying to generate an slug from null string.

I think the problem relies in the value of the form not being sending to the controller, when I change ```name```in html everything works well.

I tested using HasTranslation from Backpack instead of Spatie, but since Tag Model is extending from Spatie various methods are incompatible

**Comment by jcastroa87** [comment]
Hi.

Due to not activity on many years i will close this discussion, if there is needed feel free to reopen or create a new one.

Cheers.
